src/csvw/dsv_dialects.py

```diff
@@ -9,14 +11,25 @@
 }
 
 
+# FIXME: replace with attrs.validators.ge(0) from attrs 21.3.0
 def _non_negative(instance, attribute, value):
-    if value < 0:
+    if value < 0:  # pragma: no cover
         raise ValueError('{0} is not a valid {1}'.format(value, attribute.name))
 
 
 non_negative_int = [attr.validators.instance_of(int), _non_negative]
 
 
+def convert_encoding(s):
+    s = utils.converter(str, 'utf-8', s)
+    try:
+        _ = 'x'.encode(ENCODING_MAP.get(s, s))
+        return s
+    except LookupError:
+        warnings.warn('Invalid value for property: {}'.format(s))
+        return 'utf-8'
```

Comment:
The CSVW spec requires a warning to be emitted when an invalid encoding is specified.